packages/project-account-switcher/src/behavior/ProjectAccountSwitcherBehavior.js

```diff
@@ -0,0 +1,117 @@
+import { Component } from "react";
+import PropTypes from "prop-types";
+
+/* eslint-disable no-else-return */
+export default class ProjectAccountSwitcherBehavior extends Component {
+  static propTypes = {
+    /** List of Accounts */
+    accounts: PropTypes.arrayOf(
+      PropTypes.shape({
+        id: PropTypes.string,
+        image: PropTypes.string,
+        label: PropTypes.string
+      })
+    ),
+    /** Currently selected Account */
+    activeAccount: PropTypes.shape({
+      id: PropTypes.string,
+      image: PropTypes.string,
+      label: PropTypes.string
+    }),
+    /** Currently selected Project */
+    activeProject: PropTypes.shape({
+      id: PropTypes.string,
+      image: PropTypes.string,
+      label: PropTypes.string
+    }),
+    /** Render prop */
+    children: PropTypes.func,
+    /** Initially sets the selected Account, but allows user action to change it */
+    onClick: PropTypes.func,
+    /** Called when a user clicks on the target element */
+    onTargetClick: PropTypes.func,
+    /** Opens the Flyout */
+    open: PropTypes.bool,
+    /** List of Projects */
+    projects: PropTypes.arrayOf(
+      PropTypes.shape({
+        id: PropTypes.string,
+        image: PropTypes.string,
+        label: PropTypes.string
+      })
+    )
+  };
+
+  static defaultProps = {
+    open: false
+  };
+
+  state = {
+    activeAccount: this.props.activeAccount,
+    activeProject: this.props.activeProject,
+    open: this.props.open
+  };
+
+  getAccount(id) {
+    if (this.props.accounts) {
+      return this.props.accounts.filter(account => account.id === id)[0];
+    } else {
+      return null;
+    }
+  }
+
+  getProject(id) {
+    if (this.props.projects) {
+      return this.props.projects.filter(project => project.id === id)[0];
+    } else {
+      return null;
+    }
+  }
+
+  handleTargetClick = event => {
+    const { onTargetClick } = this.props;
+
+    if (onTargetClick) {
+      onTargetClick(event);
+    }
+
+    this.setState({
+      open: !this.state.open
+    });
+  };
+
+  handleAccountClick = event => {
+    const { onClick } = this.props;
+
+    if (onClick) {
+      onClick(event);
+    }
+
+    this.setState({
+      activeAccount: this.getAccount(event.target.dataset.accountId)
```

Comment:
Typically we want to avoid putting data related to business logic in the DOM like this, and if we need to, we should make the property more unique like `higPasAccountId`.

The preferred method to do this in React is to create event handlers for each component. An example can be seen [here](https://github.com/Autodesk/hig/blob/7e7c9b7ba7a44775b783f2351aa11bbe9e272da1/packages/notifications-flyout/src/NotificationsFlyout.js#L23). If you'd like to go into a deep dive on a topic related to this [read this](https://cdb.reacttraining.com/react-inline-functions-and-performance-bdff784f5578).